tests/ui/const_prop/apfloat-f64-roundtrip.rs

```diff
@@ -0,0 +1,9 @@
+// run-pass
+// compile-flags: -O -Zmir-opt-level=3 -Cno-prepopulate-passes
+// min-llvm-version: 16.0 (requires APFloat fixes in LLVM)
```

Comment:
It is still a mystery to me why with `-Cno-prepopulate-passes` this require apfloat fixes. :shrug:

Reply:
Yeah, I'm not sure either. Investigating more into LLVM at this point didn't seem worthwhile to me if it's been fixed in LLVM 16 but I could be convinced otherwise 🙂 

Perhaps the issue is that LLVM 14's bitcode reader fails to roundtrip the float?

Reply:
LLVM constant-folding is not a pass, but rather happens when instructions are attempted to be created.

There are passes, however, that do non-trivial simplifications, and propagate constants from e.g. loads of constant globals (though that might be in constant-folding, too? unsure), which re-triggers the constant-folding logic that first happened when the instructions were built, to account for their new inputs (you can almost think of it as "replace all uses of X with Y" simply rebuilding all instructions using X, and then the constructors of those instructions see Y now, which lets them try constant-folding again, and maybe get farther this time).

In fewer words: attempting to add a runtime LLVM `Instruction` to a `BasicBlock` may result in a `Constant` being produced instead (I don't know if there's a way to bypass this).

Reply:
Small addendum: you may see something different when testing *calls* (of either LLVM intrinsics or `libm` functions it knows about): IIRC, those *are* handled by a pass, it's only the native LLVM IR operations that get constant-folded as they're built.